responsibleai/responsibleai/rai_insights/rai_base_insights.py

```diff
@@ -0,0 +1,317 @@
+# Copyright (c) Microsoft Corporation
+# Licensed under the MIT License.
+
+"""Defines the RAIBaseInsights class."""
+
+import json
+import pickle
+import warnings
+from abc import ABC, abstractmethod
+from pathlib import Path
+from typing import Any, Optional
+
+import pandas as pd
+
+from responsibleai._internal.constants import Metadata
+
+_DATA = 'data'
+_DTYPES = 'dtypes'
+_TRAIN = 'train'
+_TEST = 'test'
+_MODEL = Metadata.MODEL
+_MODEL_PKL = _MODEL + '.pkl'
+_SERIALIZER = 'serializer'
+_MANAGERS = 'managers'
+_JSON_EXTENSION = '.json'
+
+
+class RAIBaseInsights(ABC):
+    """Defines the base class RAIBaseInsights for the top-level API.
+
+    This class is abstract and should not be instantiated.
+    """
+
+    def __init__(self, model: Any, train: pd.DataFrame, test: pd.DataFrame,
```

Comment:
Shouldn't it be `model: Optional[Any]` for things like causal analysis?

Reply:
great point, updated!